src/components/Tabs/view/Tabs.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FC, useCallback } from 'react';
+import { TabListProps, TabsProps, HandleTabsClick } from '../types';
+import { useFilteredChildrenByKey } from '../hooks/useFilteredChildrenByKey';
+
+export const Tabs = <T extends TabListProps>({
+    List,
+    activeKey,
+    onClick,
+    children,
+}: TabsProps<T>): ReturnType<FC<TabsProps<T>>> => {
+    const child = useFilteredChildrenByKey(children, activeKey);
+    //TODO: type error :{}
```

Comment:
To jsem nacastil na `any` vždycky, viz `wrap` nebo jakékoliv HoC které jsem kdy psal

Tady je možná řešení, neměl jsem čas ani chuť to číst upřímně :D
https://github.com/microsoft/TypeScript/issues/29873